src/porepy/applications/convergence_analysis.py

```diff
@@ -577,3 +583,92 @@ def l2_error(
             raise ZeroDivisionError("Attempted division by zero.")
 
         return numerator / denominator
+
+    @staticmethod
+    def lp_norm(
+        vec: np.ndarray,
+        integration_weights: Optional[pp.number | np.ndarray] = None,
+        p: pp.number = 2,
+    ) -> float:
+        r"""The discrete :math:`L_p`-norm of a vector or function evaluated on a
+        discretized grid.
```

Comment:
```suggestion
        r"""The discrete :math:`L_p`-norm of a vector or function, with a given set of weights.
```
This is not confined to a grid only.